exercises/retree/retree_test.py

```diff
@@ -0,0 +1,59 @@
+import unittest
+
+from retree import treeFromTraversals
+
+
+# Tests adapted from `problem-specifications//canonical-data.json` @ v1.0.0
+
+class TwoFerTest(unittest.TestCase):
+    def test_empty_tree(self):
+        preorder = []
+        inorder = []
+        expected = {}
+        self.assertEqual(treeFromTraversals(preorder, inorder), expected)
+
+    def test_tree_with_one_item(self):
+        preorder = ["a"]
+        inorder = ["a"]
+        expected = {"v": "a", "l": {}, "r": {}}
+        self.assertEqual(treeFromTraversals(preorder, inorder), expected)
+
+    def test_tree_with_many_items(self):
+        preorder = ["a", "i", "x", "f", "r"]
+        inorder = ["i", "a", "f", "x", "r"]
+        expected = {"v": "a", "l": {"v": "i", "l": {}, "r": {}},
+                    "r": {"v": "x", "l": {"v": "f", "l": {}, "r": {}},
+                          "r": {"v": "r", "l": {}, "r": {}}}}
```

Comment:
I've been thinking about this formatting a bit... for more complex trees, it is helpful to expand the tree a bit for readability:

```Python
expected = {
  "v": "a",
  "l": {
    "v": "i",
    "l": {},
    "r": {},
  },
  "r": {
    "v": "x",
    "l": {
      "v": "f",
      "l": {},
      "r": {},
    },
    "r": {
      "v": "r",
      "l": {},
      "r": {}
    }
  }
}
```